proxy/http/HttpTransact.cc

```diff
@@ -288,6 +288,8 @@ find_server_and_update_current_info(HttpTransact::State* s)
     /* fall through */
   default:
     update_current_info(&s->current, &s->server_info, HttpTransact::ORIGIN_SERVER, (s->current.attempts)++);
+    // TS-2157 must update port here because we know struct has been set
+    s->current.server->dst_addr.port() = s->hdr_info.server_request.port_get();
```

Comment:
this is kind of a hack-- before the port value was an int and initialized in the transact separately from the port in addr. this no longer works getting rid of the int port so had to add this here after we update the info.